cmd/root.go

```diff
@@ -167,14 +143,20 @@ var rootCmd = &cobra.Command{
 
 				if strings.ToLower(response) != "y\n" {
 					color.Green("Generated anchored Dockerfile\n")
-					fmt.Println(builder.String())
+					nodes.Write(os.Stdout)
 					return fmt.Errorf("exiting without writing file")
 				}
 			}
-			err = os.WriteFile(outputName, []byte(builder.String()), 0600)
+
+			f, err := os.Create(filepath.Clean(outputName))
```

Comment:
Probably better to give this a  more description variable name rather than `f` — similarly with `w`